### PR TITLE
hachimarupop: init at 2022.07.12

### PR DESCRIPTION
**Pull request by KiaraGrouwstra**
## Description of changes

Adding the HachiMaruPop font from https://github.com/noriokanisawa/HachiMaruPop.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by WolfangAukang** [comment]
Please consider adding the package to pkgs/by-name instead of the current directory - see [info](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md)

File: pkgs/by-name/ha/hachimarupop/package.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "hachimarupop";
+  version = "2022.07.12";
```
Comment (WolfangAukang) [inline pkgs/by-name/ha/hachimarupop/package.nix:8]:
```suggestion
  version = "unstable-2022-07-12";
```

**Review by WolfangAukang — CHANGES_REQUESTED** [review]
Only change to consider. Rest LGTM

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 267767` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hachimarupop</li>
  </ul>
</details>

File: pkgs/by-name/ha/hachimarupop/package.nix (hunk 1 above)
Comment (KiaraGrouwstra) [reply pkgs/by-name/ha/hachimarupop/package.nix]: thanks, i pushed to reflect that.

**Review by WolfangAukang — APPROVED** [review]

File: pkgs/by-name/ha/hachimarupop/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    runHook preInstall
+    install -Dm444 fonts/ttf/*.ttf -t $out/share/fonts/truetype/
+    runHook postInstall
+  '';
+
+  meta = {
+    homepage = "https://github.com/noriokanisawa/HachiMaruPop";
+    description = "A cute, Japanese font";
+    license = lib.licenses.ofl;
+    maintainers = with lib.maintainers; [ AndersonTorres ];
```
Comment (WolfangAukang) [inline pkgs/by-name/ha/hachimarupop/package.nix:27]: Quick question, is this intended? As this is a new package PR, I think you should set yourself as the maintainer

Comment (KiaraGrouwstra) [reply pkgs/by-name/ha/hachimarupop/package.nix:27]: @WolfangAukang i see.
what happened was, i was seeing a merge conflict here listing this particular maintainer (and altered package name using dashes), figured someone else might have edited my PR (not sure how), and rolled with it, accepting what i considered to be their changes.
i'm open to changing it again tho.

Comment (WolfangAukang) [reply pkgs/by-name/ha/hachimarupop/package.nix:27]: No problem, that makes it clear. Checking it thoroughly, someone packaged it and got merged before yours (https://github.com/NixOS/nixpkgs/pull/267922).

**Comment by WolfangAukang** [comment]
Closing PR because https://github.com/NixOS/nixpkgs/pull/267922 has been merged

**Pull request closed without merging**